### PR TITLE
option to disable fused adam

### PR DESCRIPTION
**Pull request by justusschock**

### CODE REVIEW

File: lightning_mingpt/models.py [hunk 1]
```diff
@@ -115,18 +112,30 @@ def generate(self, idx, max_new_tokens, temperature=1.0, do_sample=False, top_k=
 
 class DeepSpeedGPT(GPT):
     # TODO: activation checkpointing (requires overriding forward)
-    def __init__(self, offload=False, **kwargs):
+    def __init__(self, fused_adam: bool = True, offload: bool = False, **kwargs):
+        if fused_adam and offload:
+            raise RuntimeError('Cannot use FusedAdam on cpu! Please choose one of them.')
```
Comment (lantiga) [inline lightning_mingpt/models.py:117]:
```suggestion
            raise RuntimeError('Cannot use FusedAdam on cpu! Please set `fused_adam=False` or set `accelerator="gpu"` in the Trainer.')
```

**Review by lantiga — APPROVED** [review]
Tweaked error message

File: lightning_mingpt/models.py (hunk 1 above)
Comment (justusschock) [reply lightning_mingpt/models.py:117]: @lantiga setting `accelerator="gpu"` does not resolve this. You can still use the CPUAdam from Deepspeed as this one offloads the optimizer state. The point of this error message is that you can't use both at the same time so you need to decide for the fused cuda version or the CPU offload.

Comment (lantiga) [reply lightning_mingpt/models.py:117]: Can you reword the error message so that it's clearer? It is not clear what "them" refers to.

Comment (justusschock) [reply lightning_mingpt/models.py:117]:
```suggestion
            raise RuntimeError('Cannot use FusedAdam and CPUAdam at the same time! Please set either `fused_adam` or `offload` to False.')
```

**Pull request merged**